addons/l10n_tw_edi_ecpay/tests/test_edi.py

```diff
@@ -0,0 +1,338 @@
+            [{
+                'l10n_tw_edi_ecpay_invoice_id': 'AB11100099',
+                'l10n_tw_edi_invoice_create_date': datetime.strptime('2025-01-06 15:00:00', '%Y-%m-%d %H:%M:%S'),
+                'l10n_tw_edi_state': 'valid',
+            }]
+        )
+
+        self.assertTrue(self.basic_invoice.l10n_tw_edi_file_id)
+
+    def test_03_failed_submission(self):
+        """
+        This test will test a flow where the submission itself (not the documents inside) fails for any reason.
+        A general error as such should be handled, but is not expected and should be treated as a bug on our side.
+
+        As we submit a single invoice, we expect a UserError to be raised.
+        """
+        send_and_print = self.create_send_and_print(self.basic_invoice)
+        with patch(CALL_API_METHOD, new=self._test_03_mock):
+            with self.assertRaises(UserError):
+                send_and_print.action_send_and_print()
+
+    @freeze_time("2025-01-06 15:00:00")
+    def test_04_invalid_invoice(self):
+        send_and_print = self.create_send_and_print(self.basic_invoice)
+        with patch(CALL_API_METHOD, new=self._test_04_mock):
+            send_and_print.action_send_and_print()
+            wizard_vals = {'journal_id': self.basic_invoice.journal_id.id, 'reason': 'refund'}
+            wizard_reverse = self.env['account.move.reversal']\
+                .with_context(active_ids=self.basic_invoice.id, active_model='account.move').create(wizard_vals)
+            wizard_reverse.reverse_moves(is_modify=True)
+        self.assertRecordValues(
+            self.basic_invoice,
+            [{
+                'l10n_tw_edi_state': 'invalid',
+            }]
+        )
+
+    @freeze_time("2025-01-06 15:00:00")
+    def test_05_refund_invoice(self):
+        send_and_print = self.create_send_and_print(self.basic_invoice)
+        with patch(CALL_API_METHOD, new=self._test_05_mock):
+            send_and_print.action_send_and_print()
+            wizard_vals = {'journal_id': self.basic_invoice.journal_id.id}
+            wizard_reverse = self.env['account.move.reversal']\
+                .with_context(active_ids=self.basic_invoice.id, active_model='account.move').create(wizard_vals)
+            wizard_reverse.reverse_moves(is_modify=False)
+            credit_note = wizard_reverse.new_move_ids
+            credit_note.action_post()
+            send_and_print_credit_note = self.create_send_and_print(credit_note)
+            send_and_print_credit_note.action_send_and_print()
+        self.assertEqual(credit_note.l10n_tw_edi_refund_agreement_type, wizard_reverse.l10n_tw_edi_refund_agreement_type)
+        self.assertEqual(credit_note.reversed_entry_id, self.basic_invoice)
+        self.assertEqual(credit_note.l10n_tw_edi_ecpay_invoice_id, self.basic_invoice.l10n_tw_edi_ecpay_invoice_id)
+        self.assertEqual(credit_note.l10n_tw_edi_invoice_create_date, self.basic_invoice.l10n_tw_edi_invoice_create_date)
```

Comment:
use assertRecordValues here.
Also, you should set the new fields in the wizard to be sure it affects the credit note as expected.